examples/voice-assistant/log_transcriptions.py

```diff
@@ -0,0 +1,70 @@
+import asyncio
+
+from dotenv import load_dotenv
+from livekit import rtc
+from livekit.agents import AutoSubscribe, JobContext, WorkerOptions, cli, llm
+from livekit.agents.voice_assistant import VoiceAssistant
+from livekit.plugins import deepgram, openai, silero
+
+load_dotenv()
+
+
+async def entrypoint(ctx: JobContext):
+    initial_ctx = llm.ChatContext().append(
+        role="system",
+        text=(
+            "You are a voice assistant created by LiveKit. Your interface with users will be voice. "
+            "You should use short and concise responses, and avoiding usage of unpronouncable punctuation."
+        ),
+    )
+
+    await ctx.connect(auto_subscribe=AutoSubscribe.AUDIO_ONLY)
+
+    assistant = VoiceAssistant(
+        vad=silero.VAD.load(),
+        stt=deepgram.STT(),
+        llm=openai.LLM(),
+        tts=openai.TTS(),
+        chat_ctx=initial_ctx,
+    )
+    assistant.start(ctx.room)
+
+    # listen to incoming chat messages, only required if you'd like the agent to
+    # answer incoming messages from Chat
+    chat = rtc.ChatManager(ctx.room)
+
+    async def answer_from_text(txt: str):
+        chat_ctx = assistant.chat_ctx.copy()
+        chat_ctx.append(role="user", text=txt)
+        stream = assistant.llm.chat(chat_ctx=chat_ctx)
+        await assistant.say(stream)
+
+    @chat.on("message_received")
+    def on_chat_received(msg: rtc.ChatMessage):
+        if msg.message:
+            asyncio.create_task(answer_from_text(msg.message))
+
+    # clear file before writing to it
+    with open("transcriptions.log", "w"):
+        pass
+
+    @assistant.on("user_speech_committed")
+    def on_user_speech_committed(msg: llm.ChatMessage):
+        # convert string lists to strings, drop images
+        if isinstance(msg.content, list):
+            msg.content = "\n".join(
+                "[image]" if isinstance(x, llm.ChatImage) else x for x in msg
+            )
+        with open("transcriptions.log", "a+") as f:
```

Comment:
let's avoid blocking ops.
we could also just get the assistant.chat_ctx in a shutdown callback and write everything at once?

Reply:
hmm I think we should not assume that everything will be buffered. it's better to demonstrate control where user may choose how to buffer

Reply:
so should we have both a shutdown log and a real-time log?

Reply:
for example, what if I'd like to send transcriptions to my backend service on the fly to log? (in order to improve latency and minimize risk of losing data). 

for really long sessions (think 2h+) we also don't want to buffer everything in memory.

ideally we'd also expose timing data (by emitting them in realtime, we can basically avoid having to tag timestamps)

additionally, some users may use summarization techniques with the LLM, so the chatcontext isn't an accurate representation of each sentence being said by the user/agent.

Reply:
new commit reverts the changes, going back to realtime, and also makes it non-blocking.
i also print the current datetime in the header for each message.